app/models/request_information_type.rb

```diff
@@ -1,6 +1,41 @@
+# frozen_string_literal: true
+
+# A request information type associates a pipline with request information,
+# either via {Request::Metadata} or a {LabEvent}
```

Comment:
Could you explain this a bit more? I don't think I've come across this area of the data model before. It looks to me like Request Information Type Id is a foreign key on the 'Request Information' table... but there are no rows in that table. Was it deprecated? How does Request Information Type link to {Request::Metadata} and {LabEvent}? I see you've added methods below that look like they pull information from those tables, but what was this object's purpose before you added those methods?

Reply:
I was a bit new to request information types before this as well.

`RequestInformation` is before my time, and I'm not sure about the decisions around it. I'm not sure if it got deprecated, or never really got used. I believe it essentially lost the battle to request metadata, although given some of its functionality also ended up in LabEvents, I'm not entirely certain what happened. I do know that one of the former team leads, Lars, was generally speaking in favour of metadata tables over key value pairs, but I'm not really certain why. (Partly I think as I fall in the opposite camp)

So RequestInformationTypes appear to be left over from this battle between request-metadata and the lab-events. I'm not sure if its a unification that never happened, or a split that occurred early in its development. As a result, they end up looking in one place, and then the other.

Before my changes it still did an almost identical thing, its just all the associated logic was contained on Request. I flipped it around because:
1) Request already has a lot on its plate, whereas this is almost exclusively concerned with fetching and displaying data
2) The methods on request seemed to talk to the request information type more than they did the request.

Reply:
OK thanks for the background :)